demos/jans-chip/android/app/src/main/java/io/jans/chip/services/PlayIntegrityService.java

```diff
@@ -0,0 +1,214 @@
+            Log.e("Integrity token Obtained result", "failure");
+            //Toast.makeText(context, "Integrity token Obtained result :: failure", Toast.LENGTH_SHORT).show();
+            dbH.addAppIntegrity(setErrorInAppIntegrityObj("Error in obtaining integrity token :: " + e.getMessage()));
+            throw new RuntimeException(e);
+        });
+    }
+
+    private class getTokenResponse extends AsyncTask<String, Integer, String[]> {
+
+        private boolean hasError = false;
+        private DBHandler dbH;
+
+        public getTokenResponse(DBHandler dbH) {
+            this.dbH = dbH;
+        }
+
+        @Override
+        protected String[] doInBackground(String... token) {
+
+            OkHttpClient client = new OkHttpClient();
+            Request request = new Request.Builder()
+                    .get()
+                    .url(AppConfig.INTEGRITY_APP_SERVER_URL + "/api/check?token=" + token[0])
+                    .build();
+
+            Response response = null;
+            try {
+                response = client.newCall(request).execute();
+
+
+                if (!response.isSuccessful()) {
+                    hasError = true;
+                    Log.e("Response from App server :: Unsuccessful, Response code :: ", String.valueOf(response.code()));
+                    //Toast.makeText(context, "Response from App server :: Unsuccessful, Response code :: " + response.code(), Toast.LENGTH_SHORT).show();
+                    dbH.addAppIntegrity(setErrorInAppIntegrityObj("Error in obtaining response from aap server :: Response code :: " + response.code()));
+                    return new String[]{"Api request error", "Response code: " + response.code()};
+                }
+                ResponseBody responseBody = response.body();
+
+                if (responseBody == null) {
+                    hasError = true;
+                    Log.e("Response from App server :: ", "Response body is empty");
+                    //Toast.makeText(context, "Api request error :: Empty response", Toast.LENGTH_SHORT).show();
+                    dbH.addAppIntegrity(setErrorInAppIntegrityObj("Empty response obtained from App Server."));
+                    return new String[]{"Api request error", "Empty response"};
+                }
+
+                JSONObject json = null;
+                json = new JSONObject(responseBody.string());
+
+
+                if (json.has("error")) {
+                    hasError = true;
+                    Log.e("Response from App server :: ", "Response body has error");
+                    //Toast.makeText(context, "Api request error :: " + json.getString("error"), Toast.LENGTH_SHORT).show();
+                    dbH.addAppIntegrity(setErrorInAppIntegrityObj("Response from App server has error :: " + json.getString("error")));
+                    return new String[]{"Api request error", json.getString("error")};
+                }
+
+                if (!json.has("appIntegrity")) {
```

Comment:
Please use decodable object there instead of has("data_key")

Reply:
Now using DTO to get appIntegrity from API.